indexing-service/src/main/java/org/apache/druid/indexing/common/config/TaskConfig.java

```diff
@@ -184,14 +228,48 @@ public String getBaseDir()
   }
 
   @JsonProperty
-  public File getBaseTaskDir()
+  public List<StorageLocationConfig> getBaseTaskDirLocations()
+  {
+    return baseTaskDirLocations;
+  }
+
+  public List<StorageLocationConfig> getBaseTaskDirChildrenLocations(String child)
+  {
+    return getBaseTaskDirLocations().stream()
+                                    .map(location -> new StorageLocationConfig(
+                                        getChildFileFromLocation(location, child),
+                                        location.getMaxSize(),
+                                        location.getFreeSpacePercent()))
+                                    .collect(Collectors.toList());
+  }
+
+  private File getChildFileFromLocation(StorageLocationConfig location, String child)
+  {
+    return new File(location.getPath(), child);
+  }
+
+  private StorageLocationConfig getBaseTaskDirLocationForId(String taskId)
+  {
+    if (taskId == null) {
+      return null;
+    }
+    // TODO: consider the size of each of the base locations for distribution
+    // use uniformly RANDOM assignment
+    int index = taskId.hashCode() % baseTaskDirLocations.size();
+    if (index < 0) {
+      index = -(index + 1);
+    }
+    return baseTaskDirLocations.get(index);
+  }
+
+  public File getTaskBaseDir(String taskId)
   {
-    return baseTaskDir;
+    return getBaseTaskDirLocationForId(taskId).getPath();
   }
 
   public File getTaskDir(String taskId)
   {
-    return new File(baseTaskDir, taskId);
+    return new File(getTaskBaseDir(taskId), taskId);
```

Comment:
From the code above, the `getTaskBaseDir` can return null. This would cause the ctor of `File` throw NPE.

Reply:
Sorry, could you please explain how?
The list will have at least one value, and I'll be returning a File corresponding to one of those for a given task

Reply:
`getTaskBaseDir` checks if the `taskId` is null. And if it's null, a null is returned and will be passed to the ctor of `File` which then throws an NPE.

Usually, we check the arguments at the public methods level. So, if the `taskId` could be null, it's better to check it in this function.

Reply:
Ah yes, sorry. I'll add the check